include/aspect/mesh_deformation/fastscapecc_adapter.h

```diff
@@ -0,0 +1,390 @@
+/*
+  Copyright (C) 2011 - 2024 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _aspect_mesh_deformation_fastscapecc_adapter_h
+#define _aspect_mesh_deformation_fastscapecc_adapter_h
+
+// #ifndef FASTSCAPELIB_DEALII_GRID_H_
+// #define FASTSCAPELIB_DEALII_GRID_H_
+
+#include <unordered_map>
+#include <vector>
+
+// Include the Xtensor configuration file so that we can query the
+// Xtensor version number below to address the fact that between
+// 0.25.0 and 0.26.0, all Xtensor include files were moved around,
+// see https://github.com/xtensor-stack/xtensor/pull/2829 and the
+// complaints therein.
+#if __has_include(<xtensor/core/xtensor_config.hpp>)  // 0.26.0 and later
+#  include <xtensor/core/xtensor_config.hpp>
+#else // 0.25.0 and earlier
+#  include <xtensor/xtensor_config.hpp>
+#endif
+
+#if XTENSOR_VERSION_MAJOR ==0 && XTENSOR_VERSION_MINOR <= 25
+#  include <xtensor/xbroadcast.hpp>
+#  include <xtensor/xmath.hpp>
+#  include <xtensor/xarray.hpp>
+#  include <xtensor/xtensor.hpp>
+#else
+#  include <xtensor/views/xbroadcast.hpp>
+#  include <xtensor/core/xmath.hpp>
+#  include <xtensor/containers/xarray.hpp>
+#  include <xtensor/containers/xtensor.hpp>
+#endif
+
+#include <fastscapelib/grid/base.hpp>
+
+// Up until FastScape++ 0.2.2, there was a file xtensor_utils.hpp that declared
+// alias types that are used elsewhere. After that, the file went away, see
+//   https://github.com/fastscape-lem/fastscapelib/pull/155
+// but we still need the contents and so have to declare them ourselves.
```

Comment:
Fastscapelib's flexible container interface has indeed changed since v0.2.2, but it hasn't been released yet.

Wouldn't be easier to stick with the latest stable release of Fastscapelib, and then update the code here (or in a follow-up PR) to use the new container interface once v0.3.0 is released?

Since the integration between ASPECT is not yet finalized (and will likely be very new at the time when Fastscapelib v0.3.0 will be released - hopefully very shortly), I guess that it is fine to require the last released version as a minimum working version in ASPECT?